src/util/language-syntax-errors.ts

```diff
@@ -0,0 +1,63 @@
+// import { FaUnlink, FaRandom } from 'react-icons/fa';
+import { FaExclamationTriangle } from 'react-icons/fa';
+import * as monaco from 'monaco-editor/esm/vs/editor/editor.api';
+
+export enum SyntaxError {
+  Generic = 'generic',
+  MissingInitializer = 'init-missing',
+  TypeMissmatch = 'type-missmatch',
+}
+
+export type Highlight = {
+  startLine: number,
+  startColumn: number,
+  endLine: number,
+  endColumn: number
+}
+
+export type CadenceSyntaxError = {
+  message: string,
+  type: string,
+  position: monaco.IPosition,
+  icon: any,
+  highlight: Highlight
+}
+
+export const formatMarker = (marker: monaco.editor.IMarker): CadenceSyntaxError => {
+  return {
+    message: marker.message,
+    type: SyntaxError.Generic,
+    icon: FaExclamationTriangle,
+    position: {
+      lineNumber: marker.startLineNumber,
+      column: marker.startColumn
+    },
+    highlight: {
+      startLine: marker.startLineNumber,
+      endLine: marker.endLineNumber,
+      startColumn: marker.startColumn,
+      endColumn: marker.endColumn
+    }
+  }
+};
+
+export const getIconByErrorType = (errorType: SyntaxError): any => {
+  switch (errorType) {
+    /*
+    case SyntaxError.TypeMissmatch:
+      return FaRandom;
+    case SyntaxError.Generic:
+      return FaExclamationTriangle;
+    */
```

Comment:
For now all of these are errors. I don't think the PG should have specific knowledge of different specific errors, this will become a maintenance nightmare (errors are not a public interface yet – if we start depending on them outside of Cadence, we have to keep them ideally backwards-compatible).

We do have the distinction between errors and warnings though. Can you get that from the marker?